Google.PowerShell.IntegrationTests/CloudResourceManager/CloudResourceManager.IamPolicy.Tests.ps1

```diff
@@ -2,19 +2,117 @@
+        { Get-GcIamPolicyBinding -Project "asdf" } | Should Throw "403"
     }
-}
\ No newline at end of file
+}
+
+Describe "Add-GcIamPolicyBinding" {
+    It "should work for -User" {
+        $role = "roles/browser"
+        $user = "quoct@google.com"
+
+        try {
+            Add-GcIamPolicyBinding -User $user -Role $role -Project $gcloudPowerShellProject2
+
+            $bindings = Get-GcIamPolicyBinding -Project $gcloudPowerShellProject2
+            $binding = $bindings | Where-Object {$_.Role -eq $role}
+            $binding.Members -contains "user:$user" | Should Be $true
+        }
+        finally {
+            gcloud projects remove-iam-policy-binding $gcloudPowerShellProject2 `
+                                                      --member="user:$user" --role="$role" 2>$null
+        }
+    }
+
+    It "should work for -Group" {
+        $role = "roles/logging.viewer"
+        $group = "cloudsharp-eng@google.com"
+
+        try {
+            Add-GcIamPolicyBinding -Group $group -Role $role -Project $gcloudPowerShellProject2
+
+            $bindings = Get-GcIamPolicyBinding -Project $gcloudPowerShellProject2
+            $binding = $bindings | Where-Object {$_.Role -eq $role}
+            $binding.Members -contains "group:$group" | Should Be $true
+        }
+        finally {
+            gcloud projects remove-iam-policy-binding $gcloudPowerShellProject2 `
+                                                      --member="group:$group" --role="$role" 2>$null
+        }
+    }
+
+    It "should work for -ServiceAccount" {
+        $role = "roles/datastore.viewer"
+
+        try {
+            Add-GcIamPolicyBinding -ServiceAccount $appveyorServiceAccEmail -Role $role -Project $gcloudPowerShellProject2
+            $bindings = Get-GcIamPolicyBinding -Project $gcloudPowerShellProject2
+            $binding = $bindings | Where-Object {$_.Role -eq $role}
+            $binding.Members -contains "serviceAccount:$appveyorServiceAccEmail" | Should Be $true
+        }
+        finally {
+            gcloud projects remove-iam-policy-binding $gcloudPowerShellProject2 `
+                                                      --member="serviceAccount:$appveyorServiceAccEmail" --role="$role" 2>$null
+        }
+    }
+
+    It "should work for -Domain" {
+        $role = "roles/viewer"
+        $domain = "google.com"
+
+        try {
+            Add-GcIamPolicyBinding -Domain $domain -Role $role -Project $gcloudPowerShellProject2
+            $bindings = Get-GcIamPolicyBinding -Project $gcloudPowerShellProject2
+            $binding = $bindings | Where-Object {$_.Role -eq $role}
+            $binding.Members -contains "domain:$domain" | Should Be $true
+        }
+        finally {
+            gcloud projects remove-iam-policy-binding $gcloudPowerShellProject2 `
+                                                      --member="domain:$domain" --role="$role" 2>$null
+        }
+    }
+
+    It "should not remove existing bindings" {
```

Comment:
Maybe it should be:
It "should not overwrite similar existing bindings" {

Something to convey that if you have a role binding and add another role binding, you now have two instead of one new one.